LIIKUNTA-18 | Add administrative division filter to search page

Jokinen opened the pull request:
Once fetching for administrative divisions work, this change can be tested and merged.

Jokinen commented:
Resolved merge conflicts and improved i18n support.

tuovinensanttu commented on src/components/search/searchPageSearchForm/searchPageSearchForm.module.scss:
```diff
@@ -8,6 +8,15 @@
+    }
+  }
+
+  .submitButton {
+    grid-column: 4;
```
This should be inside the breakpoint definition above. As it is now, it will break the mobile layout


![image](https://user-images.githubusercontent.com/25960123/130204816-4c702957-299f-4d86-b69f-54c983c00658.png)

tuovinensanttu commented on src/widgets/administrativeDivisionDropdown/AdministrativeDivisionDropdown.tsx:
```diff
@@ -0,0 +1,86 @@
+  ];
+  const defaultValue = options.find((option) => option.value === value);
+
+  return (
+    <Select
+      {...delegated}
+      defaultValue={defaultValue}
+      label={label}
+      clearButtonAriaLabel="Poista kaikki valinnat"
+      selectedItemRemoveButtonAriaLabel="Poista ${value}"
+      options={options}
+      onChange={handleOnChange}
+      placeholder={placeholder}
+      icon={<IconLocation />}
+    />
+  );
```
There is open discussion in [slack](https://helsinkicity.slack.com/archives/C01DWA87NJC/p1629444430026600), that the component should be similar to one tapahtumat.hel.fi uses. Based on that this component should be `Combobox` instead of `Select`. Ofc we can disable the multiselect option for the time being, as US doesn't support it yet.

Jokinen commented:
Fixed the mobile layout issue @tuovinensanttu mentioned and changed the input into a combobox.

I also sorted and filtered the returned divisions based on the discussion on Slack. Perhaps this behaviour will move into the search backend at some point at least partially, but I thought that the fastest way to a working implementation was to use a temporary change in the UI layer.

Jokinen commented:
Integrated the division control into the map view as well by sharing filtering logic between the list and map views. Also modified styles slightly by removing `overflow-y: hidden` so that the division dropdown can overlap the search header.

A few venues looked to be missing locations so I added logic into the map view which avoids rendering an item if it does not have a location.

tuovinensanttu reviewed (approved):
Looks good to me 👍

Jokinen commented:
Rebased to resolve conflicts.
